src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+            throw new DataProcessingException("Couldn't get cars from DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversByCarId(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverQuery =
+                "SELECT cars.id AS id, cars.model, cars.manufacturer_id, "
+                        + "manufacturers.name, manufacturers.country "
+                        + "FROM cars_drivers JOIN cars ON id = cars_drivers.car_id "
+                        + "JOIN manufacturers ON cars.manufacturer_id "
+                        + "WHERE cars.is_deleted = FALSE AND cars_drivers.driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement =
+                        connection.prepareStatement(getAllByDriverQuery)) {
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromDb(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from a DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversByCarId(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement =
+                        connection.prepareStatement(updateCarQuery)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car in DB: " + car, e);
+        }
+        deleteDriverFromCar(car);
+        addDriverToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarByIdQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarByIdStatement =
+                        connection.prepareStatement(deleteCarByIdQuery)) {
+            deleteCarByIdStatement.setLong(1, id);
+            int updatedRows = deleteCarByIdStatement.executeUpdate();
+            return updatedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete a car from DB by id: " + id, e);
+        }
+    }
+
+    private void insertDriver(Car car) {

```

Comment:
what is the difference between this method and `addDriverToCar()`? Let's keep it DRY